Fix bug: GHDL crash when asked to analyse directory instead of file.
Debian bug #573885.
GHDL GNA bug #21308.

Patch by Jonas Baggett.
Source: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=573885

Index: ghdl-0.30~svn20130213/ghdl/files_map.adb
===================================================================
--- ghdl-0.30~svn20130213.orig/ghdl/files_map.adb	2013-12-08 16:26:51.050533000 +0100
+++ ghdl-0.30~svn20130213/ghdl/files_map.adb	2013-12-08 16:26:57.138533000 +0100
@@ -757,6 +757,9 @@
       declare
          Filename : String := Get_Pathname (Directory, Name, True);
       begin
+         if not Is_Regular_File(Filename'Address) then
+            return No_Source_File_Entry;
+         end if;
          Fd := Open_Read (Filename'Address, Binary);
          if Fd = Invalid_FD then
             return No_Source_File_Entry;
